src/store-tree/index.ls

```diff
@@ -40,45 +37,34 @@ class StoreTree extends StoreNode
 
 
   $is-loading: ->
-    for own key of @_children when @[key].$is-loading!
-      return yes
-    no
+    @_children |> values |> any (.$is-loading!)
 
 
   $reset: !-> @$batch-emit-updates ~>
-    @_for-each-subnode (subnode) -> subnode.$reset!
+    @_children |> Obj.each (.$reset!)
 
 
   $set: (data) !-> @$batch-emit-updates ~>
     switch typeof! data
-    | \Object   => @_for-each-subnode (subnode, key) -> subnode.$set data[key] if data[key] isnt undefined
-    | \Null     => @_for-each-subnode (subnode) -> subnode.$set null
+    | \Object   => @_children |> obj-to-pairs |> each ([key, subnode]) -> subnode.$set data[key] if data[key] isnt undefined
+    | \Null     => @_children |> Obj.each (.$set null)
     | otherwise => throw Error 'calling $set on a tree must be called with an object or null'
 
 
   $set-loading: (loading = yes) !-> @$batch-emit-updates ~>
-    @_for-each-subnode (subnode) -> subnode.$set-loading loading
+    @_children |> Obj.each (.$set-loading loading)
 
 
   $set-error: (err) !~> @$batch-emit-updates ~>
-    @_for-each-subnode (subnode) -> subnode.$set-error err
+    @_children |> Obj.each (.$set-error err)
 
 
   $get: ->
-    obj = {}
-    @_for-each-subnode (subnode, key) -> obj[key] = subnode.$get!
-    obj
+    @_children |> Obj.map (.$get!)
```

Comment:
Sounds good. I was using the list map at one point but it appears I no longer am